presto-main/src/main/java/com/facebook/presto/operator/SpillableGroupedTopNBuilder.java

```diff
@@ -265,7 +262,9 @@ public WorkProcessor.TransformationState<WorkProcessor<Page>> process(Optional<P
                 if (inputIsPresent) {
                     Page inputPage = inputPageOptional.get();
                     boolean done = outputInMemoryGroupedTopNBuilder.processPage(inputPage).process();
-                    verify(done);
+                    if (!done) {
+                        return blocked(memoryWaitingFutureSupplier.get());
```

Comment:
so this wasn't causing the OOM issues, right because we would have failed.  it was just an additional thing that needed to be fixed (though also related to memory management)